examples/e02_transparent_guild_sharding/src/main.rs

```diff
@@ -25,7 +26,7 @@ struct Handler;
 #[async_trait]
 impl EventHandler for Handler {
     async fn message(&self, ctx: Context, msg: Message) {
-        if msg.content == "!ping" {
+        if &*msg.content == "!ping" {
```

Comment:
Agh, you're right that `Box<str>` doesn't have everything we want and doesn't always auto-deref. This sucks, I don't know what to do. Maybe a followup PR can introduce a newtype or something that implements what we want. Or we could revert some of the fields and keep them as `String` for the sake of ergonomics. I'd say `Message::content` is a good candidate for that.